### PR TITLE
v766 init

### PR DESCRIPTION
**Pull request by OmniacDev**
Started working on v766, getting a bit carried away with other stuff though, so just PR'ing this now rather than later

### CODE REVIEW

File: crates/proto/src/version/mod.rs [hunk 1]
```diff
@@ -5,3 +5,4 @@ pub mod v686;
 pub mod v712;
 pub mod v729;
 pub mod v748;
```
Comment (theaddonn) [inline crates/proto/src/version/mod.rs:7]: are there any stable versions between `v766` and `v748`?

File: crates/proto/src/version/v671/helper.rs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+use crate::helper::ProtoHelper;
+use crate::version::v729::gamepackets::GamePackets;
```
Comment (theaddonn) [inline crates/proto/src/version/v671/helper.rs]: are you sure that is the correct import?

File: crates/proto/src/version/v671/helper.rs [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+pub struct ProtoHelperV671;
```
Comment (theaddonn) [inline crates/proto/src/version/v671/helper.rs:4]: isnt this PR for `766` and not `671`?

**Comment by theaddonn** [comment]
Generally is this PR for v766 or v671 and also.. are we now doing small steps at a time between v662 and v671 etc or one big step, since I think doing multiple small steps might be more efficient and faster?

File: crates/proto/src/version/mod.rs (hunk 1 above)
Comment (OmniacDev) [reply crates/proto/src/version/mod.rs:7]: Nope, v766 is the latest stable, for some reason they haven't pulled it into the proto docs yet though

File: crates/proto/src/version/v671/helper.rs (hunk 3 above)
Comment (OmniacDev) [reply crates/proto/src/version/v671/helper.rs:4]: Started doing v671 before just doing v766

File: crates/proto/src/version/mod.rs (hunk 1 above)
Comment (theaddonn) [reply crates/proto/src/version/mod.rs:7]: Okay good to know!

File: crates/proto/src/version/v671/helper.rs (hunk 2 above)
Comment (OmniacDev) [reply crates/proto/src/version/v671/helper.rs]: missed that, should be fixed now for them (v662 was also wrong lol)

**Comment by OmniacDev** [comment]
What do you mean by small steps?

**Comment by theaddonn** [comment]
> What do you mean by small steps?

Instead of making the gigantic jump from v662 to v766, we could make smaller steps v662 -> v671 -> v685 -> v686 (literally accidentally released by mojang lol) ->v712 -> v729 -> v748 -> 766

We could control these smaller steps better and the likelihood of missing out on something is smaller. We could also use the provided change logs by Mojang to make this easier. This would require a tiny bit more effort but it might be worth it. In the meanwhile I still need to fix some bugs regarding packet encoding/decoding and after that we could start working on the main server. We do not directly need the newest version for that since our design allows us to upgrade easily, so prototyping on an older version is fine...

**Comment by OmniacDev** [comment]
Well I was gonna finish v766 just so we have latest done, and then work backwards, since the version "cascade" (dunno if that's the right word), is the same either way. Pretty much if I finish v766, I can go back and do v748, where probably most of the changes in v766 will just be moved into, and then repeat for every version down. 

Also, the official changelogs are terrible. I'm just creating a PR from the different branches in my proto docs fork and then just using the diff to see what actually has changed.

**Comment by theaddonn** [comment]
fair enough. We will continue with implementing v766 then! Thanks :)

**Review by theaddonn — APPROVED** [review]

**Pull request merged**